blog/views.py

```diff
@@ -0,0 +1,32 @@
+from django.http import HttpResponse, HttpRequest, HttpResponseRedirect
+from django.shortcuts import render
+from django.urls import reverse
+from django.views import generic
+
+from blog.forms import CommentaryForm
+from blog.models import Post, Commentary
+
+
+class PostListView(generic.ListView):
+    model = Post
+    queryset = Post.objects.select_related("owner").prefetch_related(
+        "commentaries"
+    )
+    paginate_by = 5
+
+
+def post_detail_view(request: HttpRequest, pk: int) -> HttpResponse:
+    user = request.user
+    post = Post.objects.prefetch_related("commentaries__user").get(id=pk)
```

Comment:
It's a good practice to handle the case when the post with the given id does not exist. You can use Django's get_object_or_404 function to automatically return a 404 response if the Post does not exist.